src/lib/pickles/step.ml

```diff
@@ -843,6 +843,9 @@ struct
                               ~public_input:public_inputs
                           with
                           | None ->
+                              if
+                                Option.is_some (Sys.getenv_opt "ERROR_ON_PROOF")
+                              then failwith "Regenerated proof" ;
```

Comment:
I understand it might be a quick hack. However. "ERROR_ON_PROOF=false" and "ERROR_ON_PROOF=true" are not distinguished at this level. It might cause some unexpected incomprehensions.

I suggest defining a more comprehensive function
```
let is_error_on_proof_set () = 
  match Sys.getenv_opt "ERROR_ON_PROOF" with 
  | Some ("true"  | "t" (* insert whatever value is okay here *)) -> true 
  | None | Some _ -> false 
```
and use that instead.

Reply:
If we're aftercode  nitpicking, we should use `Sys.getenv` and match the exception in the pattern matching above.

Reply:
I only really care about the CI speed improvements here. Nobody should ever use this, expect when following the instructions in the `README.md`